$OpenBSD: patch-src_lib_asiolink_tcp_socket_h,v 1.1 2018/01/31 22:29:09 rsadowski Exp $

fix build with boost 1.66
- use native_handle() for getting native socket type
commit 17ed090beb6cefa78e29a95c645eb6da4edf8

Index: src/lib/asiolink/tcp_socket.h
--- src/lib/asiolink/tcp_socket.h.orig
+++ src/lib/asiolink/tcp_socket.h
@@ -75,7 +75,11 @@ class TCPSocket : public IOAsioSocket<C> { (public)
 
     /// \brief Return file descriptor of underlying socket
     virtual int getNative() const {
+#if BOOST_VERSION < 106600
         return (socket_.native());
+#else
+        return (socket_.native_handle());
+#endif
     }
 
     /// \brief Return protocol of socket
